libs/langchain/langchain/embeddings/huggingface.py

```diff
@@ -279,9 +279,15 @@ class HuggingFaceInferenceAPIEmbeddings(BaseModel, Embeddings):
     """Your API key for the HuggingFace Inference API."""
     model_name: str = "sentence-transformers/all-MiniLM-L6-v2"
     """The name of the model to use for text embeddings."""
+    api_url: Optional[str] = None
```

Comment:
out of curiosity, will this always end in model_name like default api url or is that not guaranteed?

Reply:
i think the main question for @dmitryrPlanner5D is whether this is really a template or a full url?

if a full url, then model_name ends up not really being used? so may be best to make it a template? but not sure if that meets @dmitryrPlanner5D motivations for opening this pr

Reply:
My motivation was to add support of models not only publicly deployed (and having url format of https://api-inference.huggingface.co/pipeline/feature-extraction/{self.model_name}), but also protected endpoints which you can create on hugging face which look like this: https://q2qi18cil57jg1g5.eastus.azure.endpoints.huggingface.cloud (not having model name in it).

So answering your question `api_url` intended to be a full url, not a template.